src/reducers/selectionReducer.js

```diff
@@ -35,6 +35,10 @@ const initialState = {
 };
 
 export default (state=initialState, { id: cardId, type }) => {
+  if(!cardId) {
```

Comment:
Sure thing.

As it was before, there was no validation on which actions would trigger the selection reducer.  This caused `undefined` to be returned by the state machine, which was then set into the state:

![](https://ameo.link/u/4f5.png)

This change basically just prevents the reducer from being triggered if there is no `id` attribute on the action.